DiscordBot.Core/Interfaces/IRandomResponseProvider.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace DiscordBot.Core.Interfaces
+{
+    public interface IRandomResponseProvider
```

Comment:
`IRandomFactProvider` and `IRandomFactProvider` are literally the same. Same issue goes to all their implementations. Extract all the similarities to one place. Also the names are misleading - they don't provide _random_ text, just allow to get all of the saved values and add new ones.